include/wx/generic/dirctrlg.h

```diff
@@ -215,12 +254,18 @@ class WXDLLIMPEXP_CORE wxGenericDirCtrl: public wxControl
 
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_CORE, wxEVT_DIRCTRL_SELECTIONCHANGED, wxTreeEvent );
 wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_CORE, wxEVT_DIRCTRL_FILEACTIVATED, wxTreeEvent );
+wxDECLARE_EXPORTED_EVENT( WXDLLIMPEXP_CORE, wxEVT_DIRCTRL_MENU_POPPED_UP, wxCommandEvent );
```

Comment:
I think the event type should end with `-ING` suffix to indicate that it is sent before the action takes place, according to another convention (there are several event pairs like `CHANGING`/`CHANGED`, `SELECTING`/`SELECTED`).

So my suggestion would be to call this one `wxEVT_DIRCTRL_SHOWING_POPUP_MENU`.

And I also wonder why we shouldn't use `wxTreeEvent` for it too, if only for consistency? This would at least give the associated tree item in the event itself, which would seem to be potentially useful.